### ISSUE TITLE
Machine file as a config file

### ISSUE THREAD

**Issue by danielabdi-noaa** [ufs-community/ufs-srweather-app#386]
## Description
TLDR: The machine files under `ush/machine` are confusing in that in most cases they provide default values and in some cases override values for a specific machine. As a result, the user may not know where to put a variable ( `config.yaml` or `hera.sh`) to set it properly. Also, some machine files such as `noaacloud.sh` exploit the freedom that a shell script gives to do something that is best left for modulefiles. So it is best to make machine files strictly config files that either override or provide default values for a specific machine, most likely with the following purpose and order:

- **config_defaults.yaml**: generic default values
- **machine.yaml**: machine specific default values
- **config.yaml**: user specified values that override the above two

### Details
---
`hera.sh` has this which provides default value
```
NCORES_PER_NODE=${NCORES_PER_NODE:-40} 
```
Later there is this which overrides a variable
```
RUN_CMD_UTILS="srun"
```
So specifying `RUN_CMD_UTILS` in `config.yaml` will not have any effect since the machine file will override it anyway, while `NCORES_PER_NODE` is not overridden.

---
Another confusing aspect is that when running WE2E tests the machine file is sourced and used to construct `config.yaml`, while on the other hand, ex-scripts for tasks such as `run_fcst` source the machine file later. This can be avoided by constructing `var_defns` file once including contributions from all 3 config files in the order specified above.

---
The `noaacloud.sh` machine file has also things like this that are best left for modulefiles
```
export PROJ_LIB=/contrib/GST/miniconda/envs/regional_workflow/share/proj
export OPT=/contrib/EPIC/hpc-modules
export PATH=${PATH}:/contrib/GST/miniconda/envs/regional_workflow/bin
```
And the below is a variable used only in `run_WE2E`
```
export build_mod_fn="wflow_noaacloud"
```
Later there is this 
```
. /contrib/EPIC/.bash_conda
```
I am not sure sourcing the noaacloud machine file from python works properly because of this.

## Solution
Convert machine files to yaml files and use it to update  `config_defaults.yaml` once during workflow generation, just like `config.yaml` does, and write the files values in `var_defns.sh` for use by later tasks. Will need to decide whether to use them for providing default values or overriding values, but the first one sounds better. Then, get rid of all `source MACHINE_FILE` statements from ex-script of tasks and `run_we2e`. This will give clarilty to the purpose of machine files the and prevent abuse of the freedom that the current machine shell scripts provide.

## Requirements**
None

## Acceptance Criteria (Definition of Done)
Machine files converted to yaml config files and used to update `config_defaults.yaml` once during workflow generation. Note that most of what is defined in machine files is already in `config_defaults.yaml` so it makes sense they are treated as config files. Find alternate solution for the problems discussed above for `noaacloud.sh` specifically.

## Dependencies (optional)
None

## Alternative Solutions (optional)
Do nothing
